lib/ibc-solidity/src/lib.rs

```diff
@@ -0,0 +1,629 @@
+                bytes calldata data
+            ) external returns (uint64);
+
+            function recvPacket(
+                MsgPacketRecv calldata msg_
+            ) external;
+
+            function recvIntentPacket(
+                MsgIntentPacketRecv calldata msg_
+            ) external;
+
+            function writeAcknowledgement(
+                Packet calldata packet,
+                bytes memory acknowledgement
+            ) external;
+
+            function acknowledgePacket(
+                MsgPacketAcknowledgement calldata msg_
+            ) external;
+
+            function timeoutPacket(
+                MsgPacketTimeout calldata msg_
+            ) external;
+
+            function batchSend(
+                MsgBatchSend calldata msg_
+            ) external;
+
+            function batchAcks(
+                MsgBatchAcks calldata msg_
+            ) external;
+
+            // IBC MODULE
+
+            error ErrClientNotFound();
+            error ErrModuleNotFound();
+            error ErrInvalidConnectionState();
+            error ErrInvalidChannelState();
+
+            // IBC CLIENT
+
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ClientRegistered(string clientType, address clientAddress);
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ClientCreated(string clientType, uint32 clientId);
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ClientUpdated(uint32 clientId, uint64 height);
+
+            error ErrClientTypeAlreadyExists();
+            error ErrClientTypeNotFound();
+
+            // IBC CONNECTION
+
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ConnectionOpenInit(
+                uint32 connectionId, uint32 clientId, uint32 counterpartyClientId
+            );
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ConnectionOpenTry(
+                uint32 connectionId,
+                uint32 clientId,
+                uint32 counterpartyClientId,
+                uint32 counterpartyConnectionId
+            );
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ConnectionOpenAck(
+                uint32 connectionId,
+                uint32 clientId,
+                uint32 counterpartyClientId,
+                uint32 counterpartyConnectionId
+            );
+            #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+            event ConnectionOpenConfirm(
+                uint32 connectionId,
+                uint32 clientId,
+                uint32 counterpartyClientId,
+                uint32 counterpartyConnectionId
+            );
+
+            // error ErrInvalidProof();
```

Comment:
remove plis

Reply:
will clean up this crate in a followup PR